src/etc/inc/openvpn.inc

```diff
@@ -1238,8 +1240,24 @@ function openvpn_restart($mode, $settings) {
 	/* start the new process */
 	$fpath = $g['varetc_path']."/openvpn/{$mode_id}.conf";
 	openvpn_clear_route($mode, $settings);
-	mwexec("/usr/local/sbin/openvpn --config " . escapeshellarg($fpath));
-
+	$res = mwexec("/usr/local/sbin/openvpn --config " . escapeshellarg($fpath));
+	if ($res == 0) {
+		$i = 0;
+		$pid = "--";
+		while ($i < 3000) {
+			if (file_exists($pfile) ) {
+				$pid = rtrim(file_get_contents($pfile));
+				if (is_numericint($pid)) {
+					break;
+				}
+			}
+			usleep(1000);
+			$i++;
```

Comment:
Run through this loop 3000 times with usleep(1000) inside looks like too much to me, it can run for a looooong time.

Other than that, isvalidpid() does basically the same operation you are doing and it also checks if process is running. Any reason for not using it?